### PR TITLE
Clean dead code System.Net*

### PR DESCRIPTION
**Pull request by MisinformedDNA**
Contributes to #17905

### CODE REVIEW

File: src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs [hunk 1]
```diff
@@ -2,14 +2,14 @@
...
 
-using Microsoft.Win32.SafeHandles;
 using System;
 using System.Collections.Generic;
 using System.Net;
 using System.Net.Sockets;
 using System.Runtime.InteropServices;
 using System.Security.Authentication.ExtendedProtection;
 using System.Threading;
+using Microsoft.Win32.SafeHandles;
```
Comment (davidsh) [inline src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs]: Why is this line moving?  We normally keep things in alphabetical order.

File: src/System.Net.Mail/tests/Unit/EncodedStreamFactoryTest.cs [hunk 2]
```diff
@@ -2,47 +2,13 @@
...
 
-using System.IO;
 using System.Text;
 using Xunit;
 
 namespace System.Net.Mime.Tests
 {
     public class EncodedStreamFactoryTests
     {
-        [Fact]
```
Comment (davidsh) [inline src/System.Net.Mail/tests/Unit/EncodedStreamFactoryTest.cs]: Why are you deleting tests?  Tests should not be deleted.

File: src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs (hunk 1 above)
Comment (MisinformedDNA) [reply src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs]: Oh, that's default behavior of VS Power Tools. I can change that back.

Comment (Clockwork-Muse) [reply src/Common/src/Interop/Windows/HttpApi/Interop.HttpApi.cs]: For the curious, the rule being observed here is "`System` first", which may be something we should observe.

File: src/Common/src/System/Net/WebHeaderEncoding.cs [hunk 3]
```diff
@@ -75,57 +75,5 @@ internal static unsafe void GetBytes(string myString, int charIndex, int charCou
                 }
             }
         }
-        internal static unsafe byte[] GetBytes(string myString)
```
Comment (rmkerr) [inline src/Common/src/System/Net/WebHeaderEncoding.cs]: This is used here:
https://github.com/dotnet/corefx/blob/5a37188238e90d8566e6161b43ac483a671bb7d2/src/System.Net.HttpListener/src/System/Net/Managed/HttpListenerResponse.Managed.cs#L281

File: src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs [hunk 4]
```diff
@@ -283,16 +283,6 @@ public override string ToString()
             }
         }
 
-        internal string GetHeaderString(string name)
```
Comment (rmkerr) [inline src/System.Net.Http/src/System/Net/Http/Headers/HttpHeaders.cs]: Used here:
https://github.com/dotnet/corefx/blob/5a37188238e90d8566e6161b43ac483a671bb7d2/src/System.Net.Http/src/System/Net/Http/CurlHandler/CurlHandler.EasyRequest.cs#L881

File: src/Common/src/System/Net/Internals/SocketExceptionFactory.cs [hunk 5]
```diff
@@ -18,12 +18,6 @@ public ExtendedSocketException(int errorCode, EndPoint endPoint)
                 _endPoint = endPoint;
             }
 
-            public ExtendedSocketException(SocketError socketError, int platformError)
```
Comment (rmkerr) [inline src/Common/src/System/Net/Internals/SocketExceptionFactory.cs]: On Unix this is also used here:
https://github.com/dotnet/corefx/blob/5a37188238e90d8566e6161b43ac483a671bb7d2/src/Common/src/System/Net/Internals/SocketExceptionFactory.Unix.cs#L13

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj [hunk 6]
```diff
@@ -209,4 +206,4 @@
     <Reference Include="System.Threading.Tasks" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
-</Project>
+</Project>
```
Comment (caesar1995) [inline src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: The newline character should not be removed.

File: src/System.Net.NetworkInformation/src/System.Net.NetworkInformation.csproj [hunk 7]
```diff
@@ -374,4 +365,4 @@
     <Reference Include="System.Threading.Timer" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
-</Project>
+</Project>
```
Comment (caesar1995) [inline src/System.Net.NetworkInformation/src/System.Net.NetworkInformation.csproj]: Same here.

File: src/System.Net.Primitives/src/System.Net.Primitives.csproj [hunk 8]
```diff
@@ -209,4 +203,4 @@
     <Reference Include="System.Threading" />
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
-</Project>
+</Project>
```
Comment (caesar1995) [inline src/System.Net.Primitives/src/System.Net.Primitives.csproj]: Same.

File: src/System.Net.Mail/tests/Unit/EncodedStreamFactoryTest.cs (hunk 2 above)
Comment (MisinformedDNA) [reply src/System.Net.Mail/tests/Unit/EncodedStreamFactoryTest.cs]: These tests were removed because the relevant methods were removed [here](https://github.com/dotnet/corefx/pull/29302/files#diff-f106cd330f5aea61da67184a76a26d0cL21)
and [here](https://github.com/dotnet/corefx/pull/29302/files#diff-d3c7d3b0ebd5e40867f4e209450c47d7L14). These were removed in accordance with #17905. (See http://tempcoverage.blob.core.windows.net/report3/System.Net.Mail.diff.html)

File: src/System.Net.NameResolution/src/System.Net.NameResolution.csproj (hunk 6 above)
Comment (MisinformedDNA) [reply src/System.Net.NameResolution/src/System.Net.NameResolution.csproj]: That's VS at work. Is there an issue filed with VS to stop removing the last line? Or at least be consistent?

**Comment by MisinformedDNA** [comment]
Huh, last time I looked, I thought there were only timeout issues. Anyway...

I checked the code and the errors don't make any sense to me. The first error says `The type or namespace name 'SafeHandleZeroOrMinusOneIsInvalid' could not be found`, but I didn't remove that type. 

Just to be sure, I just pulled `master`, merged branch, ran `clean.cmd` and finally `build.cmd -allConfigurations` and got no errors.

What would cause the build to fail in the cloud, but succeed locally? Any ideas are welcome.

**Comment by stephentoub** [comment]
> but I didn't remove that type

No, but you removed usings, e.g. https://github.com/dotnet/corefx/pull/29302/files#diff-566153c4d0c117b01307c758c0f737b1L5.

**Comment by MisinformedDNA** [comment]
Oh. Good eye! Should I see this build error with `build.cmd -allConfigurations`? Is there a way to see these builds break locally?

**Comment by stephentoub** [comment]
@weshaggard, @ericstj, in what situation would the Packaging All Configurations build succeed but other legs fail due to product compilation errors, as happened here?

**Comment by stephentoub** [comment]
Does --allConfigurations not build release?

File: src/Common/tests/Tests/System/StringExtensions.Tests.cs [hunk 9]
```diff
@@ -44,7 +44,7 @@ public void SubstringTrim_VariousInputsOutputs(string source, int startIndex, in
 
             if (length == source.Length - startIndex)
             {
-                validate(source.SubstringTrim(startIndex));
+                validate(source.SubstringTrim(startIndex, length));
             }
```
Comment (stephentoub) [inline src/Common/tests/Tests/System/StringExtensions.Tests.cs]: This whole if block can be deleted.... you removed the overload it was testing.

File: src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.Auth.cs [hunk 10]
```diff
@@ -73,10 +70,6 @@ internal void ParseExtensions(string[] extensions)
                 {
                     _dsnEnabled = true;
                 }
-                else if (string.Compare(extension, 0, "STARTTLS", 0, 8, StringComparison.OrdinalIgnoreCase) == 0)
-                {
-                    _serverSupportsStartTls = true;
-                }
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.Auth.cs]: Why did you remove this?

File: src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.Auth.cs [hunk 11]
```diff
@@ -16,7 +16,6 @@ internal partial class SmtpConnection
     {
         private bool _serverSupportsEai;
         private bool _dsnEnabled;
-        private bool _serverSupportsStartTls;
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.Auth.cs]: Why did you remove this?

File: src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs [hunk 12]
```diff
@@ -231,13 +224,10 @@ internal void GetConnection(string host, int port)
 
             if (_enableSsl)
             {
-                if (!_serverSupportsStartTls)
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs]: Why did you remove this?

File: src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs [hunk 13]
```diff
@@ -425,7 +415,8 @@ internal static void End(IAsyncResult result)
 
             internal void GetConnection()
             {
-                if (NetEventSource.IsEnabled) NetEventSource.Enter(this);
+                if (NetEventSource.IsEnabled)
+                    NetEventSource.Enter(this);
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs]: Please undo these formatting changes.

File: src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs [hunk 14]
```diff
@@ -558,13 +551,10 @@ private bool SendEHello()
 
                     if (_connection.EnableSsl)
                     {
-                        if (!_connection._serverSupportsStartTls)
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs]: Why did you remove this?

File: src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs [hunk 15]
```diff
@@ -617,13 +607,10 @@ private static void SendEHelloCallback(IAsyncResult result)
 
                         if (thisPtr._connection.EnableSsl)
                         {
-                            if (!thisPtr._connection._serverSupportsStartTls)
```
Comment (stephentoub) [inline src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.cs]: Why did you remove this?

File: src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.Auth.cs (hunk 11 above)
Comment (MisinformedDNA) [reply src/System.Net.Mail/src/System/Net/Mail/SmtpConnection.Auth.cs]: Per #17905 and http://tempcoverage.blob.core.windows.net/report3/System.Net.Mail.diff.html, `internal bool ServerSupportsStartTls { get; }` was to be removed as dead code.

That led to a build error:
> D:\Repos\corefx\src\System.Net.Mail\src\System\Net\Mail\SmtpConnection.Auth.cs(19,22): error CS0414: The field 'SmtpConnection._serverSupportsStartTls' is assigned but its value is never used [D:\Re
pos\corefx\src\System.Net.Mail\tests\Unit\System.Net.Mail.Unit.Tests.csproj]

I saw "assigned but its value is never used" and immediately thought I could delete it. I didn't notice that this error actually came up only for the test project.

Fixing...

**Comment by MisinformedDNA** [comment]
Not entirely sure why OSX failed. Timeout?

**Comment by stephentoub** [comment]
@dotnet/dnceng, can you help me understand the situation with this OSX leg?
https://ci3.dot.net/job/dotnet_corefx/job/master/job/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/11931/
GitHub shows it as a failed leg / red X, and
https://ci3.dot.net/job/dotnet_corefx/job/master/job/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/11931/
shows `OSX.1012.Amd64.Open - x64 - Debug (98e03d54-7e49-450b-b75f-5bf156848ceb)` as a failed test run, but then going to the details page at https://mc.dot.net/#/user/MisinformedDNA/pr~2Fjenkins~2Fdotnet~2Fcorefx~2Fmaster~2F/test~2Ffunctional~2Fcli~2F/c6fc19441c30d6ede85372fd9477e053456071a3 says there are 0 failures, and it does list `1` for `PassOnRetry`.
How am I supposed to interpet this?

**Comment by Chrisboh** [comment]
Created https://github.com/dotnet/core-eng/issues/3405 to take a look at this.

**Comment by MattGal** [comment]
@stephentoub I think you're good to merge;  I think there may be a bug with the API or Jenkins plugin that is preventing it from understanding PassOnRetry.

Summary:
Test 'System.Net.HttpListener.Tests' failed the first time with:
```
2018-05-01 11:20:08,637: INFO: proc(54): run_and_log_output: Output:    System.Net.Tests.InvalidClientRequestTests.GetContext_InvalidRequest_DoesNotGetContext [FAIL]
2018-05-01 11:20:08,638: INFO: proc(54): run_and_log_output: Output:       System.Net.Internals.SocketExceptionFactory+ExtendedSocketException : Connection refused 127.0.0.1:1575
2018-05-01 11:20:08,642: INFO: proc(54): run_and_log_output: Output:       Stack Trace:
2018-05-01 11:20:08,708: INFO: proc(54): run_and_log_output: Output:          /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs(4421,0): at System.Net.Sockets.Socket.DoConnect(EndPoint endPointSnapshot, SocketAddress socketAddress)
2018-05-01 11:20:08,709: INFO: proc(54): run_and_log_output: Output:          /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs(832,0): at System.Net.Sockets.Socket.Connect(EndPoint remoteEP)
2018-05-01 11:20:08,709: INFO: proc(54): run_and_log_output: Output:          /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs(947,0): at System.Net.Sockets.Socket.Connect(IPAddress[] addresses, Int32 port)
2018-05-01 11:20:08,709: INFO: proc(54): run_and_log_output: Output:          --- End of stack trace from previous location where exception was thrown ---
2018-05-01 11:20:08,709: INFO: proc(54): run_and_log_output: Output:          /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs(958,0): at System.Net.Sockets.Socket.Connect(IPAddress[] addresses, Int32 port)
2018-05-01 11:20:08,710: INFO: proc(54): run_and_log_output: Output:          /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs(902,0): at System.Net.Sockets.Socket.Connect(String host, Int32 port)
2018-05-01 11:20:08,710: INFO: proc(54): run_and_log_output: Output:          /Users/dotnet-bot/j/workspace/dotnet_corefx/master/osx-TGroup_netcoreapp+CGroup_Debug+AGroup_x64+TestOuter_false_prtest/src/System.Net.HttpListener/tests/HttpListenerFactory.cs(155,0): at System.Net.Tests.HttpListenerFactory.GetConnectedSocket()
...
```
... but passed on the subsequent auto-retry (this was enabled ~1 week ago).

**Comment by stephentoub** [comment]
Thanks, @Chrisboh and @MattGal.

File: src/Common/tests/Tests/System/StringExtensions.Tests.cs [hunk 16]
```diff
@@ -27,24 +27,16 @@ public class StringExtensionTests
...
-                    Assert.Same(string.Empty, result);
-                }
-                else if (result.Length == source.Length)
-                {
-                    Assert.Same(source, result);
-                }
-            };
-
-            validate(source.SubstringTrim(startIndex, length));
+            var result = source.SubstringTrim(startIndex, length);
```
Comment (stephentoub) [inline src/Common/tests/Tests/System/StringExtensions.Tests.cs]: Nit: var => string

**Comment by Chrisboh** [comment]
The data agrees with Matt.  The test case failed once and then passed when we auto retried it.  It looks like we might have a race condition we need to track down and will keep investigating but you should be fine to merge.

**Comment by stephentoub** [comment]
@dotnet-bot test Outerloop Windows x64 Debug Build please
@dotnet-bot test Outerloop Linux x64 Release Build please

File: src/Common/src/Interop/Windows/winhttp/Interop.winhttp_types.cs [hunk 17]
```diff
@@ -289,26 +279,12 @@ public struct WINHTTP_ASYNC_RESULT
...
         public const uint API_SEND_REQUEST = 5;
 
-        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode)]
-        public struct WINHTTP_WEB_SOCKET_ASYNC_RESULT
-        {
-            public WINHTTP_ASYNC_RESULT AsyncResult;
-            public WINHTTP_WEB_SOCKET_OPERATION Operation;
-        }
-
         public enum WINHTTP_WEB_SOCKET_OPERATION
```
Comment (davidsh) [inline src/Common/src/Interop/Windows/winhttp/Interop.winhttp_types.cs]: This enum type could probably be removed as well.  We don't use WinHTTP for WebSocket anymore (we use the managed client on top of SocketsHttpHandler.

File: src/System.Net.Mail/tests/Unit/SmtpConnectionTests/EhloParseExtensionsTest.cs [hunk 18]
```diff
@@ -113,7 +113,6 @@ public void ParseExtensions_WithNoMechanismsAdvertised_ShouldNotSupportAnyAuthTy
 
             Assert.True(_smtpConnection.DSNEnabled);
             Assert.True(_smtpConnection.ServerSupportsEai);
-            Assert.True(_smtpConnection.ServerSupportsStartTls);
```
Comment (davidsh) [inline src/System.Net.Mail/tests/Unit/SmtpConnectionTests/EhloParseExtensionsTest.cs]: Why was this deleted?  It was useful as part of the unit test which tests the various internal properties.

Comment (stephentoub) [reply src/System.Net.Mail/tests/Unit/SmtpConnectionTests/EhloParseExtensionsTest.cs]: The property itself was deleted.

Comment (davidsh) [reply src/System.Net.Mail/tests/Unit/SmtpConnectionTests/EhloParseExtensionsTest.cs]: Yes, I saw that also.  But wasn't the property valid?  And if so, was this then deleted because only the unit test was reading the property value?  And thus, no actual product code was using it?
